tests/gamma/csm_observability_test.py

```diff
@@ -175,27 +205,56 @@ def test_csm_observability(self):
         #   resource creation out of self.startTestServers()
         with self.subTest("1_run_test_server"):
             start_secs = int(time.time())
-            test_server: _XdsTestServer = self.startTestServers(
-                enable_csm_observability=True,
-            )[0]
+            test_server: _XdsTestServer = self.startTestServers()[0]
 
         with self.subTest("2_start_test_client"):
             test_client: _XdsTestClient = self.startTestClient(
                 test_server,
-                enable_csm_observability=True,
                 request_payload_size=REQUEST_PAYLOAD_SIZE,
                 response_payload_size=RESPONSE_PAYLOAD_SIZE,
             )
 
         with self.subTest("3_test_server_received_rpcs_from_test_client"):
             self.assertSuccessfulRpcs(test_client)
 
-        with self.subTest("4_query_cloud_monitoring_metrics"):
+        with self.subTest("4_export_prometheus_metrics_data"):
             logger.info(
                 "Letting test client run for %d seconds to produce metric data",
                 TEST_RUN_SECS,
             )
-            time.sleep(TEST_RUN_SECS)
+            server_prometheus_logger = PrometheusLogger(
```

Comment:
We need to check that `--collect_app_logs` is on - this is save into `self.should_collect_logs`. Otherwise we get

```py
Traceback (most recent call last):
  File "/Users/sergiitk/Development/psm-interop/tests/gamma/csm_observability_test.py", line 225, in test_csm_observability
    server_prometheus_logger = PrometheusLogger(
  File "/Users/sergiitk/Development/psm-interop/tests/gamma/csm_observability_test.py", line 156, in __init__
    log_path = k8s_runner.logs_subdir / logfile_name
  File "/Users/sergiitk/Development/psm-interop/framework/test_app/runners/base_runner.py", line 56, in logs_subdir
    raise FileNotFoundError("Log collection is not enabled.")
FileNotFoundError: Log collection is not enabled.
```

Reply:
Done